doc/horizontal_autoscaling.md

```diff
@@ -0,0 +1,173 @@
+# Design Doc: Horizontal Autoscaling
+
+## Background
+
+A PaddlePaddle training job contains several trainer instances,
+several parameter server instances, and one master instance. We would
+like to automatically scale the number of training instances and the
+number of parameter server instances to fully utilize the cluster's
+computation resources.
+
+Currently, we will only support horizontal trainer
+autoscaling. Parameter server horizontal autoscaling will be supported
+in the near future. This design doc considers both of them.
+
+[Horizontal Pod Autoscaling (HPA)](https://kubernetes.io/docs/tasks/run-application/horizontal-pod-autoscale/) is
+an autoscaling solution provided by Kuberentes, but it's not suitable
+for the training job autoscaling for the following reasons:
+
+- The training job autoscaling tries to fairly distribute the
```

Comment:
The training job autoscaling tries to :
1. do autoscale one training job, in order to accelerate the training process. Or 
2. fairly distribute the computation resources to different training jobs, in order to optimize the computation resource utilization of the **entire** cluster

Reply:
I think the goal of "do autoscale one training job" is also "fairly distribute the computation resources...", so maybe they have to be in the same paragraph?
Agree that the wording can be improved, I tried to improve the wording, please take a look :).